core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1234,8 +1234,9 @@ class ReplicaManager(val config: KafkaConfig,
                                fetchOffset: Long,
                                currentTimeMs: Long): Option[Int] = {
     partition.leaderReplicaIdOpt.flatMap { leaderReplicaId =>
-      // Don't look up preferred for follower fetches via normal replication
-      if (Request.isValidBrokerId(replicaId))
+      // Don't look up preferred for follower fetches via normal replication and
+      // don't look up preferred read replica while fetch from follower replica
+      if (Request.isValidBrokerId(replicaId) || partition.leaderIdIfLocal.isEmpty)
```

Comment:
It would be better to directly use `leaderIdIfLocal` at L1236: `partition.leaderIdIfLocal.flatMap { leaderReplicaId =>`.

Reply:
@dajac  It seems better. I have updated the code. Thanks.